### PR TITLE
Better document xfail(condition)

### LINKED ISSUE #6952: Conditional xfail
I'd like to see something like `pytest.mark.xfailif(condition)`

### Use-case:
Our lib is built on multiple operating systems. Currently one test fails on OSX due to issues outside of the library. If it would fail always, I could add `@pytest.mark.xfail()`, but it's only happening on OSX. Alternatively, I could `@pytest.mark.skipif()`. But the advantage of `xflail` is that it would notify me when the external issue is solved and I can reactivate the test. With `skipif` I would not know when the test starts working again and probably forget it.

Alternatively, is there an existing pattern I could use to get a conditional xfail.

### PR DESCRIPTION
**Pull request by timhoffm**
Closes #6952.

### CODE REVIEW

File: doc/en/skipping.rst [hunk 1]
```diff
@@ -265,6 +265,22 @@ internally by raising a known exception.
 **Reference**: :ref:`pytest.mark.xfail ref`
 
 
+``condition`` parameter
+~~~~~~~~~~~~~~~~~~~~~~~
+
+If a test is only expected to fail under certain conditions, you can pass
+that condition as first parameter:
```
Comment (bluetech) [inline doc/en/skipping.rst]:
```suggestion
If a test is only expected to fail under a certain condition, you can pass
that condition as the first parameter:
```

File: doc/en/skipping.rst [hunk 2]
```diff
@@ -265,6 +265,22 @@ internally by raising a known exception.
...
+
+.. code-block:: python
+
+    @pytest.mark.xfail(sys.platform == "windows", reason="bug in a 3rd party library")
```
Comment (bluetech) [inline doc/en/skipping.rst]:
```suggestion
    @pytest.mark.xfail(sys.platform == "win32", reason="bug in a 3rd party library")
```

**Comment by timhoffm** [comment]
I've moved the `strict` section to the end. The order now corresponds to the order of the parameters in the signature, and it also brings the sections `condition` and `reason` next to each other as desired.

**Comment by bluetech** [comment]
Thanks!

File: doc/en/skipping.rst [hunk 3]
```diff
@@ -301,7 +288,7 @@ on a particular platform:
 
 .. code-block:: python
 
-    @pytest.mark.xfail(sys.version_info >= (3, 6), reason="python3.6 api changes")
+    @pytest.mark.xfail(reason="known parser issue")
```
Comment (blueyed) [inline doc/en/skipping.rst]: This example does not match the previous sentence anymore.

Comment (nicoddemus) [reply doc/en/skipping.rst]: Thanks, https://github.com/pytest-dev/pytest/pull/6980

**Pull request merged**